kotlin-audio/src/main/java/com/doublesymmetry/kotlinaudio/players/BaseAudioPlayer.kt

```diff
@@ -143,15 +145,48 @@ abstract class BaseAudioPlayer internal constructor(private val context: Context
         exoPlayer = ExoPlayer.Builder(context).apply {
             if (bufferConfig != null) setLoadControl(setupBuffer(bufferConfig))
         }.build()
+        forwardingPlayer = createForwardingPlayer()
 
         mediaSession.isActive = true
 
-        notificationManager = NotificationManager(context, exoPlayer, mediaSession.sessionToken, notificationEventHolder)
+        val playerToUse = if (playerConfig.interceptPlayerActionsTriggeredExternally) forwardingPlayer else exoPlayer
```

Comment:
Depending on the flag, we can choose to intercept Player actions via `ForwardingPlayer`. This will intercept actions triggered by the media session and notification.